subprojects/packagefiles/openssl/generator.py

```diff
@@ -0,0 +1,74 @@
+#!/usr/bin/env python3
+
+import glob
+import os
+import shutil
+import subprocess
+
+# Change to the directory of the Bash script
+script_dir = os.path.dirname(os.path.abspath(__file__))
+os.chdir(script_dir)
+
+# Node.js version should bundle OpenSSL of matching version to one specified in wrap file
+node_version = "v19.7.0"
+openssl_version = os.environ.get("OPENSSL_VERSION")
+
+if openssl_version is None:
+    with open("../../openssl.wrap", encoding="utf-8") as wrap_file:
+        for line in wrap_file:
+            if "directory = " in line:
+                openssl_version = line.strip().split()[-1]
+                break
+
+# Clone the Node.js repository
+subprocess.run(["git", "clone", "--depth", "1", "-b", node_version, "https://github.com/nodejs/node.git"])
+
+# Remove the generated-config directory
+shutil.rmtree("generated-config", ignore_errors=True)
+
+# Change directory to Node.js deps/openssl
+os.chdir("node/deps/openssl")
+
+# Apply the patch to generate meson.build for different targets
+subprocess.run(["patch", "-u", "config/generate_gypi.pl", "-i", "../../../generate_gypi.pl.patch"])
+
+# Copy meson.build template file
+shutil.copy("../../../meson.build.tmpl", "config/")
+
+# Clone the upstream OpenSSL repository
+shutil.rmtree("openssl")
+subprocess.run(["git", "clone", "--depth", "1", "--branch", "openssl-" + openssl_version, "https://github.com/openssl/openssl.git"])
+
+# Remove the config/archs directory
+shutil.rmtree("config/archs")
+
+# Run the make command with LANG set to C
+os.environ['LANG'] = 'C'
+subprocess.run(["make", "-C", "config"])
```

Comment:
```suggestion
subprocess.run(["make", "-C", "config"], env={"LANG": "C"})
```

Reply:
python has find_program?

Reply:
No, but there's [`shutil.which`](https://docs.python.org/3/library/shutil.html?highlight=which#shutil.which), as suggested in [`subprocess.Popen`'s documentation](https://docs.python.org/3/library/subprocess.html?highlight=subprocess%20run#subprocess.Popen).

Reply:
The subprocess.Popen documentation is a bit weird except in the sense that you may want to invoke APIs which do not search PATH, i.e. direct use of `os.exec*()` functions that do not have a "p" in them. OR if by "maximum reliability" one means "reliable on non-POSIX platforms like Windows, the most common platform around". Which... fair enough lol.

I do find it somewhat baffling however that reading the subprocess.py source code it appears Python will refuse to use the optimized path of posix_spawn if given a command without a dirname (a fact totally unrelated to whether it is an absolute path).

Reply:
Anyway the only point of `find_program('make')` would be doing it ***in meson.build***, if one wanted to use meson.build to check which tools are available rather than having generator.sh fail on its own.